rpcapd/win32-svc.c

```diff
@@ -33,7 +33,6 @@
 
 #include "rpcapd.h"
 #include <signal.h>
-#define	_WINSOCKAPI_
```

Comment:
Please submit this as a separate pull request.

Reply:
This goes along with its replacement set [here](https://github.com/sgeto/libpcap/blob/rpcapd/rpcapd/CMakeLists.txt#L26)

If submitted separately, there will be a nasty warning when building rpcapd.
I personally think this shouldn't be separated.

Reply:
If the right way to prevent `winsock.h` being included by `windows.h` is to define `WIN32_LEAN_AND_MEAN`, then:

- if autotools are used to build rpcapd on Windows, it also need to be changed to define `WIN32_LEAN_AND_MEAN` on Windows, and that change can include the removal of the `#define`;

- if autotools *aren't* used to build rpcapd on Windows, then the change to remove the `#define` of `_WINSOCKAPI_` can be done, as a separate change, before submitting changes to build it with CMake, as submitting it won't break any build.

Reply:
1. it's "just" a warning. There won't be any build errors.
2. autotools can't build build rpcapd. not even on unix. it errors because it is not setup to link with threads (and crypt). 
on windows, it errors even earlier (during configure). mingw can't pass the ANSI compiler check.
3. before you fix that, I have already done quite a few things (locally) on/for the autotools build system. I will push them (one by one :-D) after my cmake changes. If that's okay.

Reply:
> autotools can't build rpcapd. not even on unix.

The OS running on my development machine is a trademark-licensed Unix, and rpcapd builds fine on it, because that OS's pthreads APIs are part of the system library (`libSystem.dylib`).  It happens to have an `-lpthreads` library, but that might just be something Apple added to help build processes that *assume* an `-lpthreads` library.  It may be that other UN\*Xes don't have an `-lpthreads`.

Reply:
I see I rushed into conclusion there. It doesn't build on my linux box

Reply:
The best way to handle that is probably to first check whether, for example, `pcap_create()` is available with the standard libraries and, if it's not, check whether it's available in libpthreads and, if it is, add that to the list of libraries with which to link (and if it isn't, fail).